heliopy/data/mms.py

```diff
@@ -7,131 +7,1781 @@
+    Data Center (SDC). Data can be found at
+
+    https://lasp.colorado.edu/mms/sdc/
+
+    The following methods must be implemented by sub-classes:
+
+    - :meth:`Downloader.intervals()`: given a time interval, this
+      method should split the interval up into sub-intervals. Each of these
+      sub-intervals corresponds directly to a single file to download, store,
+      and read in.
+    - :meth:`Downloader.local_dir()`: given an interval, returns the
+      local directory in which the file is stored.
+    - :meth:`Downloader.fname()`: given an interval, returns the
+      local filename in which the file is stored.
+    - :meth:`Downloader.download()`: given an interval, download the data for
+      that interval.
+    - :meth:`Downloader.load_local_file()`: given an interval, load the local
+      file and return a :class:`pandas.DataFrame` object containing the data.
+    """
+    def __init__(self, sc=None, instr=None, mode=None, level=None,
+                 anc_product=None,
+                 data_type='science',
+                 data_root=None,
+                 dropbox_root=None,
+                 end_date=None,
+                 files=None,
+                 mirror_root=None,
+                 offline=False,
+                 optdesc=None,
+                 site='public',
+                 start_date=None,
+                 version=None):
+        """
+        Initialization method
+
+        Parameters
+        ----------
+        sc : str, list
+            Spacecraft ID.
+        instr : str, list
+            Instrument ID.
+        mode : str, list
+            Data rate mode.
+        level : str, list
+            Data product quality level. Setting level to None, "l2", or "l3"
+            automatically sets site to "public".
+        anc_product : bool
+            Set to True if you want ancillary data. Automatically sets
+            "data_type" to "ancillary".
+        data_type : str
+            Type of information requested from the SDC:
+                "science" - Science data
+                "hk" - Housekeeping data
+                "ancillary" - attitude and emphemeris data
+                "abs_selections" - Automated burst segment selections
+                "sitl_selections" - Burst selections made by the SITL
+                "gls_selections_*" - Burst selections made by the ground loop
+        data_root : str
+            Root directory in which to download MMS data. Additional
+            directories beyond data_root are created to mimic the MMS
+            file system structure. Defaults to heliopy.config['download_dir'].
+        dropbox_root : str
+            Directory in which newly created data files are temporarily saved.
+            In the MMS data processing flow, newly created data files are
+            stored in dropbox_dir. After data file validation, they are
+            removed from dropbox_dir and placed in data_root.
+        end_date : str, `datetime.datetime`
```

Comment:
It would make sense to re-arrange so that `end_date` and `start_date` are next to each other in the parameter list (both here and in the method signature).